packages/flutter/lib/src/foundation/change_notifier.dart

```diff
@@ -17,90 +18,6 @@ import 'memory_allocations.dart';
 
 export 'dart:ui' show VoidCallback;
 
-/// An object that maintains a list of listeners.
-///
-/// The listeners are typically used to notify clients that the object has been
-/// updated.
-///
-/// There are two variants of this interface:
-///
-///  * [ValueListenable], an interface that augments the [Listenable] interface
-///    with the concept of a _current value_.
-///
-///  * [Animation], an interface that augments the [ValueListenable] interface
-///    to add the concept of direction (forward or reverse).
-///
-/// Many classes in the Flutter API use or implement these interfaces. The
-/// following subclasses are especially relevant:
-///
-///  * [ChangeNotifier], which can be subclassed or mixed in to create objects
-///    that implement the [Listenable] interface.
-///
-///  * [ValueNotifier], which implements the [ValueListenable] interface with
-///    a mutable value that triggers the notifications when modified.
-///
-/// The terms "notify clients", "send notifications", "trigger notifications",
-/// and "fire notifications" are used interchangeably.
-///
-/// See also:
-///
-///  * [AnimatedBuilder], a widget that uses a builder callback to rebuild
-///    whenever a given [Listenable] triggers its notifications. This widget is
-///    commonly used with [Animation] subclasses, hence its name, but is by no
-///    means limited to animations, as it can be used with any [Listenable]. It
-///    is a subclass of [AnimatedWidget], which can be used to create widgets
-///    that are driven from a [Listenable].
-///  * [ValueListenableBuilder], a widget that uses a builder callback to
-///    rebuild whenever a [ValueListenable] object triggers its notifications,
-///    providing the builder with the value of the object.
-///  * [InheritedNotifier], an abstract superclass for widgets that use a
-///    [Listenable]'s notifications to trigger rebuilds in descendant widgets
-///    that declare a dependency on them, using the [InheritedWidget] mechanism.
-///  * [Listenable.merge], which creates a [Listenable] that triggers
-///    notifications whenever any of a list of other [Listenable]s trigger their
-///    notifications.
-abstract class Listenable {
```

Comment:
As this is, it would be a massive breaking change. Is there a way to soften this change, allowing folks time to gracefully migrate to the new package before it is removed from the framework?

Reply:
The idea is that this is a zero change for existing Flutter users.

My proof: we have green tests! (Certainly we need more verification, but the fact that I've removed these types and all of the existing types pass is a pretty good indication.)

Reply:
> Noob question, what's the breaking change if the `foundation` library exports `Listenable` and `ValueListenable`? I would assume that customers can still get these listenable types with `import 'flutter/foundation.dart'`?

Yah, see earlier comment where I picked up on that. Sorry.

Reply:
Oh sorry, I saw that comment but misunderstood it. Ignore me 😂